msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -266,9 +273,10 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
             }
         }
 
-        private bool ContainsClaimsRequiredToStoreInCache(IdToken idToken, RequestContext requestContext)
+        private bool ContainsClaimsRequiredToStoreInCache(IdToken idToken, RequestContext requestContext, Core.Instance.AuthorityType authorityType)
         {
-            if (idToken == null || string.IsNullOrEmpty(idToken.TenantId) || string.IsNullOrEmpty(idToken.PreferredUsername))
+            //PreferredUsername is not required in B2C cache scenarios
```

Comment:
>//PreferredUsername is not required in B2C cache scenarios [](start = 12, length = 58)

Did you test this? Can you aquire a token silently?

Reply:
Yes
#Closed